node/node.go

```diff
@@ -862,6 +857,8 @@ func makeNodeInfo(
 		ID_:             nodeID,
 		Network:         chainID,
 		Version:         version.TMCoreSemVer,
+		ListenAddress:   config.P2P.ListenAddress,
```

Comment:
what is the point of having `config.P2P.ListenAddress` then if it will be overwritten in any case?

Reply:
I do not think it is overwritten. We set:

```go
n.nodeInfo.ListenAddress = n.transport.ListenAddress()
```

Not the p2p config version - that's the one we dial. Under most circumstances they will be the same.